### ISSUE TITLE
Bug - <AlertActionLink> - does not support ReactNode as its child

### ISSUE THREAD

**Issue by radekkaluzik** [patternfly/patternfly-react#9006]
**Describe the problem**
`<AlertActionLink>` which is a part of `<Alert>` component does not support `ReactNode` as its child - it only supports `children: string[]`

**How do you reproduce the problem?**
```js
<AlertActionLink>
                {/** @ts-ignore*/}
                <a
                  href="https://access.redhat.com/support/policy/updates/hybridcloud-console/lifecycle"
                  target="_blank"
                  rel="noreferrer"
                  title="Learn more link"
                >
                  Learn more
                </a>
              </AlertActionLink>
```

**Expected behavior**
Add support for `ReactNode`

**Is this issue blocking you?**
Workaround is to disable TS

**Screenshots**
N/A

**What is your environment?**
 - OS: [iOS]
 - Browser [chrome]
 - Version [112.0.5615.49]

**What is your product and what release date are you targeting?**
https://console.redhat.com/

**Any other information?**
N/A

**Comment by MariaAga** [comment]
I think the expected use of alertActionLink in this case is:
``` js
        <AlertActionLink
        component="a"
          href="https://access.redhat.com/support/policy/updates/hybridcloud-console/lifecycle"
          target="_blank"
          rel="noreferrer"
          title="Learn more link"
        >
          Learn more
        </AlertActionLink>
```

**Comment by Hyperkid123** [comment]
@MariaAga it's not really about a link specifically but about the fact that `ReactNode` is not supported as a child in the prop typings. If for whatever reason you need a nested child tree, you are unable to do so without suppressing TS type checking.

**Comment by thatblindgeye** [comment]
Closed via #9278
